cypress-io/cypress
# perf: pull log attributes from test instead of storing log attrs multiple times

Apart of https://github.com/cypress-io/cypress/issues/23391
Apart of https://github.com/cypress-io/cypress/issues/21135

While profile Cypress to determine the memory impact we have, I noticed we are collecting all of the logs by id to easily pull the log attributes when a reporter command is hovered/clicked to show the console props in the devtools console and to show the associated snapshot.

We keep a reference of all log attributes on each test as well, which we need to store on the server-side when a test navigate to a new origin (thus we update top) and have to re-normalized the test runnables, determine where we left off in the spec run and repopulate the reporter with all of the logs for the spec.

This change updates to pull these  log attributes from the test itself instead of storing these attributes again.

Some examples:
- [driver/cypress/e2e/commands/actions/click.cy.js](https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/e2e/commands/actions/click.cy.js) stored object with 3060 keys to store log attributes
- [driver/cypress/e2e/commands/navigation.cy.js](https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/e2e/commands/navigation.cy.js) stored object with 748 keys to store log attributes
- [driver/cypress/e2e/commands/actions/submit.cy.js](https://github.com/cypress-io/cypress/blob/develop/packages/driver/cypress/e2e/commands/actions/submit.cy.js) stored object with 147 keys to store log attributes

<img width="1884" alt="Screen Shot 2022-09-08 at 5 05 48 PM" src="https://user-images.githubusercontent.com/14099737/189234672-2e2f51d9-8ae9-4557-8e41-35d711f728c7.png">
<img width="1522" alt="Screen Shot 2022-09-08 at 5 06 50 PM" src="https://user-images.githubusercontent.com/14099737/189234675-0ea7cc00-effc-48d6-b403-0214112708cd.png">
<img width="1514" alt="Screen Shot 2022-09-08 at 5 03 52 PM" src="https://user-images.githubusercontent.com/14099737/189234671-077b4273-e5c9-458e-9771-75e5bde6e7c4.png">

### User facing changelog

Lump under overall perf updates to reduce memory footprint.

### Steps to test

Since this change is related to interactions with the reporter, it's best to pull this down and verify this behavior has been maintained. I've verified in several places and everything seems to check out.

### PR Tasks

- [n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
